### ISSUE TITLE
Internet Explorer Interface Errors

### ISSUE THREAD

**Issue by veilrap** [HabitRPG/habitica#2879]
On IE11:

Check boxes for dailies do not appear correctly, and it somewhat disrupts usability, ESPECIALLY on touchscreens.

Drag and drop doesn't really work at all.  Dragging appears fine, dropping causes somewhat arbitrary placement.

**Comment by deilann** [comment]
The site is not optimized for Internet Explorer. I highly suggest using another browser.

**Comment by veilrap** [comment]
That doesn't seem like a fair sentiment. It was my assumption, especially given that HabitRPG has paid subscriptions, that HabitRPG is a professional production.  

While my default browser is Chrome (not that it matters), on my touch device of choice I greatly prefer to use IE because it's simply better at touch than the other two major browsers.

**Comment by deilann** [comment]
It has nothing to do with professionalism. Because of Microsoft's refusal to render pages in a logical way, similar to other browsers, even sites like Google have had issues having their site work in Internet Explorer. I work with several professional platforms that understand that getting perfect IE compatibility is a pipe dream these days. Obviously, Habit would like to support Internet Explorer, but it can't take priority over other issues when it's a convoluted issue and is fixed by using another browser.

I feel you on not wanting to use another browser on touch, causing problems with Windows on mobile-- the site is entirely broken for me on my Nook Tablet. I'm not talking simple usability issues; I'm talking the site doesn't load.

The good news is that a Windows mobile app is in development. I'm keeping IE issue tickets open because if someone wants to address them, it would be appreciated, even if it's not high priority.

If you personally want to give developers incentive to create better IE compatibility, I highly suggest putting a bounty on this ticket.

**Comment by deilann** [comment]
tldr: The site is not optimized for IE does not mean we don't WANT the site optimized for IE; it simply means the site is not optimized for IE.

I was not trying to dismiss your issue; I was trying to let you know we understand the problem exists.

**Comment by deilann** [comment]
Also, I want to let you know that if my response seemed gruff, it is simply because I woke up to 10-15 issue tickets issue responses. I like to get to them quickly, so that you know your issue is recognized and it doesn't sit unresponded, but I do this as a volunteer. Because of that, I can't give every ticket a loving and deep response immediately, especially if I've just woken up. I'm going to add a ticket for Windows app development, so that can be bountied as well. Subscriptions currently go to cover Habit's costs-- not Open Source development.

**Comment by veilrap** [comment]
That's fair, my frustration (obviously other than the bug itself) stemmed from the fact that it seemed like it was the reverse; IE optimization being unwanted vs. Simply being low priority, which is understandable.  I've worked on a number of web development projects, and I understand the annoyance of getting things to work everywhere - squashing a bug in one browser often opens the door for bugs in the other ones.

A Windows App would certainly fix the problem in the mobile space, I was actually thinking of looking into how feasible it would be to create from the HabitRPG APIs available.

I don't normally submit bug reports, but I only recently discovered HabitRPG and I thought it sounded like a great idea, (which parallels some I'd actually thought about in my hobby time), so I was hoping that my post might help in some small way :)

Good luck getting it to work on your Nook! (assuming that's a goal at somepoint)

**Comment by veilrap** [comment]
Whoops miss clicked there.

Sorry for the confusion.

**Comment by deilann** [comment]
Yes! The issue ticket certainly does help!

And I'm glad that you understand the issue of cross-browser compatibility.

Creating a Windows app is certainly feasible; it's just substantially different from the android and iOS from what I know? Which is why it's easy to code one that can easily port to iOS and android, but one would need to be specifically made for Windows. But I know little about mobile development. The ticket I created is here: https://github.com/HabitRPG/habitrpg/issues/2885

If you want to do the app development, comment on the ticket and let others know that this is something you could do. I know there are others who want a Windows app, and if everyone put a small bounty ($5 is the minimum, I think), it could become quite a bit!

Definitely fingers crossed for the Nook, too. If I need to lie down or step away from the computer, my Nook is my go-to for internet, and internet without Habit just isn't the same.

**Comment by paglias** [comment]
@veilrap @deilann I think that supporting IE especially version 11 and future ones is  very important. While I don't use IE a lot of things have changed from the days of IE 8, 9 and all their problems. 

I'll take a look at the problems @veilrap is having but I have some problems debugging directly on IE so if you could post screenshots of the bugs you're encountering it would be very useful.

For the drag & drop it's only a touch problem or also when using the mouse?

**Comment by deilann** [comment]
Glad to hear v11 has changed since 8/9.

For reference: About half a percent of all visits in the last week were IE users and less than a quarter of a percent of new users visit the site using IE. All but 4 of them were using 10 or 11.

That small of an audience is why I feel like compatibility can't take precedence over site stability and crits, but I do agree that it's important.

**Comment by veilrap** [comment]
![checkmark displacement](https://f.cloud.github.com/assets/550772/2198107/4633e3d8-98c4-11e3-9a53-c25a7f700702.PNG)

![normal checkmark](https://f.cloud.github.com/assets/550772/2198140/87005db0-98c4-11e3-8e85-eee07a17ecd5.PNG)

![xp wrongly full](https://f.cloud.github.com/assets/550772/2198109/46405582-98c4-11e3-87d6-3e067145eb42.PNG)

Running IE11 on my desktop produces those images (the one with the correct checkmark is from Chrome), the drag and drop for mouse use in desktop IE11 seems to work fine.  I'll double check on my touch device again in a few minutes to verify the touch issues.

**Comment by veilrap** [comment]
Just tried out the touch version again, had a hard time reproducing the exact issue that happened before (the somewhat random reordering).  While testing on a touch screen, I couldn't get Chrome nor IE to correctly do reordering.  Chrome didn't seem to allow me to pick up the elements at all, whereas IE11 seemed to start to pick them up but then immediately dropped them back to the same position.

**Comment by deilann** [comment]
Is the somewhat random reordering on To-Dos or other tasks?

**Comment by veilrap** [comment]
When it happened it was in the habits column.

My guess as to what was happening - Touched, initiated drag, dragged item to location i wanted, dropped item.  Then for some reason, drop location registered incorrectly and item assumed to have been dropped at the bottom of the list.

**Comment by deilann** [comment]
kk just making sure -- reordering To-Dos can be a problem in any broswer;
known issue

**Comment by sbreiler** [comment]
Would be interested to know if the new checkmarks/bars help with the UI Problems with IE and Windows 8 Phones - anyone had a chance to test it? (I hope its okay to ask it here) @veilrap

**Comment by negue** [comment]
On Nokia 620, WP8.1 the Check-Glyph looks good.
Same on IE11 on W8.1

**Comment by veilrap** [comment]
Looks good to me as well, both on IE on a tablet and on a phone.  Thumbs up.
